src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -40,63 +39,74 @@
+		"",
+		"# With booleans",
+		"toggle gravity of player"
+})
+@Since("1.4, INSERT VERSION (booleans)")
 public class EffToggle extends Effect {
 	
 	static {
-		Skript.registerEffect(EffToggle.class, "(close|turn off|de[-]activate) %blocks%", "(toggle|switch) [[the] state of] %blocks%", "(open|turn on|activate) %blocks%");
+		Skript.registerEffect(EffToggle.class, 
+				"(close|turn off|de[-]activate) %blocks%",
+		 		"(toggle|switch) [[the] state of] %booleans/blocks%",
+		  		"(open|turn on|activate) %blocks%");
 	}
 
 	@SuppressWarnings("null")
-	private Expression<Block> blocks;
+	private Expression<?> toggledExpr;
 	private int toggle;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		blocks = (Expression<Block>) vars[0];
+		toggledExpr = (Expression<?>) vars[0];
 		toggle = matchedPattern - 1;
+		if (toggledExpr.getReturnType() == Boolean.class && !ChangerUtils.acceptsChange(toggledExpr, ChangeMode.SET, Boolean.class)) {
+			Skript.error(toggledExpr.toString(null, false) + " cannot be toggled");
+			return false;
+		}
 		return true;
 	}
 
 	@Override
-	protected void execute(final Event e) {
-		for (Block b : blocks.getArray(e)) {
-			BlockData data = b.getBlockData();
-			if (toggle == -1) {
-				if (data instanceof Openable)
-					((Openable) data).setOpen(false);
-				else if (data instanceof Powerable)
-					((Powerable) data).setPowered(false);
-			} else if (toggle == 1) {
-				if (data instanceof Openable)
-					((Openable) data).setOpen(true);
-				else if (data instanceof Powerable)
-					((Powerable) data).setPowered(true);
-			} else {
-				if (data instanceof Openable) // open = NOT was open
-					((Openable) data).setOpen(!((Openable) data).isOpen());
-				else if (data instanceof Powerable) // power = NOT power
-					((Powerable) data).setPowered(!((Powerable) data).isPowered());
+	protected void execute(Event event) {
+		ArrayList<Object> toggledValues = new ArrayList<>();
+		for (Object obj : toggledExpr.getArray(event)) {
+			if (obj instanceof Block) {
+				Block block = (Block) obj;
+				BlockData data = block.getBlockData();
+				if (toggle == 0) {
+					if (data instanceof Openable) // open = NOT was open
+						((Openable) data).setOpen(!((Openable) data).isOpen());
+					else if (data instanceof Powerable) // power = NOT power
+						((Powerable) data).setPowered(!((Powerable) data).isPowered());
+				} else {
+					if (data instanceof Openable)
+						((Openable) data).setOpen(toggle == 1);
+					else if (data instanceof Powerable)
+						((Powerable) data).setPowered(toggle == 1);
+				}
+
+				block.setBlockData(data);
+				toggledValues.add(block);
+
+			} else if (obj instanceof Boolean) {
+				toggledValues.add(!(Boolean) obj);
 			}
-			
-			b.setBlockData(data);
 		}
-	}
 
+		toggledExpr.change(event, toggledValues.toArray(), ChangeMode.SET);
```

Comment:
Should not be called when only blocks are involved, and is currently not guaranteed to have acceptChange true

Reply:
Sure, what would you suggest doing instead?

Reply:
Have a check before this to make sure at least one boolean is involved. Then, in that if statement, if the list contains a block, check acceptsChange before calling change.

Reply:
Is the last commit I made what you wanted?